src/cobra/core/reaction.py

```diff
@@ -609,13 +610,20 @@ def _update_genes_from_gpr(self, new_gene_names: Optional[Set] = None) -> None:
             model_genes = self._model.genes
             self._genes = set()
             for g_id in new_gene_names:
```

Comment:
Wait, so it will add genes even if they don't appear in the GPR? That does not seem to fit the name of the function...